app/src/main/java/org/schulcloud/mobile/controllers/main/HomeworkListFragment.kt

```diff
@@ -0,0 +1,70 @@
+package org.schulcloud.mobile.controllers.main
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.support.v7.widget.DividerItemDecoration
+import android.support.v7.widget.LinearLayoutManager
+import android.view.*
+import org.schulcloud.mobile.R
+import org.schulcloud.mobile.controllers.base.BaseFragment
+import org.schulcloud.mobile.viewmodels.HomeworkListViewModel
+import kotlinx.android.synthetic.main.fragment_homework_list.*
+import org.schulcloud.mobile.controllers.homework.HomeworkActivity
+import org.schulcloud.mobile.models.homework.HomeworkRepository
+import org.schulcloud.mobile.utils.HOST
+
+class HomeworkListFragment : BaseFragment() {
+
+    companion object {
+        val TAG: String = HomeworkListFragment::class.java.simpleName
+    }
+
+    override var url: String? = "${HOST}/homework"
+
+    private lateinit var viewModel: HomeworkListViewModel
+    private val homeworkListAdapter: HomeworkListAdapter by lazy {
+        HomeworkListAdapter{
+            startActivity(HomeworkActivity.newIntent(context!!,it))
+        }.apply {
+            emptyIndicator = empty
+        }
+    }
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+        viewModel = ViewModelProviders.of(this).get(HomeworkListViewModel::class.java)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        activity?.title = getString(R.string.homework_title)
+        return inflater.inflate(R.layout.fragment_homework_list, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        swipeRefreshLayout = swipeRefresh
+
+        viewModel.homework.observe(this, Observer {
+            homeworkListAdapter.update(it ?: emptyList())
```

Comment:
This list is sorted ascending with regard to the due date. While this makes sense for yet-to-be-completed homework, old ones still appear at the top (e.g. login with the demo-teacher, and there's a homework from 2017 at the top).

Reply:
About the required behaviour: shall the old ones be shown at the end or not shown at all?

Reply:
This is fixed now. I don't know why GitHub still doesn't display it as outdated